src/GraphQL/DataLoader/DataLoaderExtensions.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace GraphQL.DataLoader
+{
+    public static class DataLoaderExtensions
+    {
+        /// <summary>
+        /// Asynchronously load data for the provided given keys
+        /// </summary>
+        /// <param name="dataLoader">The dataloader to use</param>
+        /// <param name="keys">Keys to use for loading data</param>
+        /// <returns>
+        /// A task that will complete when the DataLoader has been dispatched,
+        /// or a completed task if the result is already cached.
+        /// </returns>
+        public static Task<T[]> LoadAsync<TKey, T>(this IDataLoader<TKey, T> dataLoader, IEnumerable<TKey> keys)
+        {
+            var tasks = new List<Task<T>>(keys.Count());
```

Comment:
I think that `keys.Count()` call is redundant and can cause the key set to be recalculated if the caller somehow calculates it.  `var tasks = new List<Task<T>>();` will be quite enough.

I'm sure this is the last my remark before merge )